django_irods/storage.py

```diff
@@ -424,7 +425,13 @@ def checksum(self, full_name):
             raise ValidationError("{} cannot be found in iRODS to retrieve "
                                   "checksum".format(obj_name))
         # remove potential '\n' from stdout
-        return stdout.rstrip('\n')
+        checksum = stdout.rstrip('\n')
+        if not checksum:
+            if force_compute:
+                raise ValidationError("checksum for {} cannot be found in iRODS".format(obj_name))
+            # checksum hasn't been computed, so force the checksum computation
+            return self.checksum(full_name, force_compute=True)
+        return checksum
```

Comment:
I don't think this recursive checksum call with force_compute flag set is needed since the ichksum call earlier is synchronous, and if it computes successfully, it should make its way in the iCAT DB, and it if fails, an exception would be raised. In either case, this recursive call is not needed I believe.

Reply:
OK. I see what you are trying to do here. Disregard my comment above then.